src/ripple/app/ledger/Ledger.cpp

```diff
@@ -914,7 +916,7 @@ Ledger::read (Keylet const& k) const
     if (! value)
         return nullptr;
     auto sle = std::make_shared<SLE>(
-        value->peekSerializer(), value->key());
+        SerialIter{value->data(), value->size()}, value->key());
```

Comment:
The important thing is that the `SLE` doesn't appear to store any references to the `SerialIter` past the execution of the ctor.  What's confusing me is what the rvalue-reference overload gets us, since it just passes the rvalue-reference to the lvalue-reference ctor overload without any transformation.